HISTORY.rst

```diff
@@ -1,31 +1,49 @@
-vNEXT
+v0.4.0, 2023-Aug-17
 -----
 
+* The Linux kernel which is supported (>= 5.10.152 5.15.76, 6.0.6, 6.1+)
+* Improved documentation and automated checks
 * Python 3 support.
   Python 2 is no longer supported.
   **NOTE:** If you are running the broker in a Python virtualenv you will have to
   rebuild the virtualenv using a Python 3 interpreter.
   This can be achieved by deleting the old virtualenv folder and recreating it.
   For recreating the virtualenv you can refer to the "Installation" section of
   the broker `installation chapter in the Tunneldigger documentation`_.
+* Added cmake buildsystem to the client.
+* Removed NAT implementation.  
+* Removed dependency on Netfilter.
 * Fixed compatibility with new Linux kernels on the broker side: New kernels
   force the l2tpv3 session ID to be unique system-wide, while old tunneldigger
   clients have a hard-coded ID of 1 for both ends of the tunnel. When a new
   client talks to a new broker, they will instead use a unique ID. Moreover,
   when the broker detects that it runs on an affected kernel, it reports maximal
   usage to old clients -- so if the client does usage-based selection, it will
   pick another broker.
-* Added cmake buildsystem to the client.
-* Added option `connection_rate_limit` to configure the delay between two
-  clients connecting.  Default is 10 (seconds).
-* Added option `pmtu`, defaults to 0 (auto-discovery). If set to a non-zero
-  value, this disables PMTU discovery and replying to client's PMTU discovery
-  probes.
+* Improved PMTU discovery to shortly after tunnel creation.
+* More session and broker information is available for the hooks.
 * Improve client behavior on broker failure, and decrease some reconnect
   timeouts.
-* Removed dependency on Netfilter.
-* Removed NAT hack.  However, this means we now require a `very recent kernel`_
-  (>= 5.2.17, 5.3.1, 5.4+).
+* Improved and more detailed logging
+* l2tp_broker.cfg confguration file changes:
+
+  * Added option ``connection_rate_limit`` to configure the delay between two
+    clients connecting.  Default is 10 (seconds).  Valid values include whole
+    seconds (10) or fractions of a second (0.2).  A value of 0 disables this
+    feature.
+  * Added options ``connection_rate_limit_per_ip_count`` and 
+    ``connection_rate_limit_per_ip_time`` to configure a connection rate limit
+    per client IP address.  If either value is 0, the functionality is disabled.
+    The default values are 0.
+  * Added hook ``broker.connection-rate-limit`` which is called when the connection
+    rate per IP address limit is exceeded
+  * Added option ``pmtu``, defaults to 0 (auto-discovery). If set to a non-zero
+    value, this disables PMTU discovery and replying to client's PMTU discovery
+    probes.
+  * Removed option ``filename``. All log message are sent to STDOUT or STDERR
+  * Removed option ``port_base``.
+  * Removed option ``namespace``.
+  * Removed option ``check_modules``.
```

Comment:
```suggestion
  * Removed option ``check_modules``. It already stopped doing anything since v0.2.0.
```

Reply:
Reworded to "No longer implemented since v0.2.0"